File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/core/aws/s3/BucketName.java
package uk.nhs.nhsx.core.aws.s3;

import uk.nhs.nhsx.core.ValueType;

public class BucketName extends ValueType<BucketName> {

    private BucketName(String value) {
        super(value);
    }

    public static BucketName of(String name) {
        return new BucketName(name);
    }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/core/aws/s3/BucketName.kt
package uk.nhs.nhsx.core.aws.s3

import dev.forkhandles.values.NonBlankStringValueFactory
import dev.forkhandles.values.StringValue

class BucketName private constructor(value: String) : StringValue(value) {
    companion object : NonBlankStringValueFactory<BucketName>(::BucketName)
}
